### DISCUSSION TITLE
Show binary as ones and zeros

### DISCUSSION THREAD

**Discussion by SevHub** [wierkstudio/ciphereditor#25] [Ideas]
As far as I can tell there is no way to show a binary output as ones and zeros. With the old version that was possible. Also I miss a simple output or viewer node, where I can pipe the output of another node, and just view it another format. 

In general I have to say that I much prefere the old site, and find it much more intuitive and powerful. The only thing missing is a saving, loading and sharing feature. Too bad the old project is abandoned in favor of the new site.

**Comment by ffraenz** [comment]
Hi there!

First, let me answer your questions:

- You can use the [Binary to text encoding](https://ciphereditor.com/explore/binary-to-text) operation to turn binary data not only into things like base64, but also into base2 (binary). The 'Encoded Data' output will be text and can flow into other operations supporting text values.
- A control in ciphereditor (the things you can toggle to edit plaintext, ciphertext, and setting values) is a combination of both a viewer and the operation setting fields in [cryptii](https://cryptii.com/). Right now, controls are very basic, coming with a plain text input for text and binary values and increment/decrement controls for numeric values. However, I plan to make them much more powerful in the future (e.g. hex editor, file upload/download, syntax highlighting, etc.). You can get a standalone control when pressing '+' and selecting 'Empty control' at the very bottom. Controls have the specialty that they become visible in the above program. You can see that, when you press the 'Up arrow' icon at the top.

Thank you for your general feedback. I'd like leave a quick note as to why the old project is currently not in active development.

The sequential world of cryptii is very accessible but also very limited. To mention a few:
- there is no way to connect/provide values other than the main brick input/output,
- there is no way to handle multiple outputs (e.g. as used for [Generate PGP Key](https://ciphereditor.com/explore/pgp-encryption) operation),
- you can't have multiple flows without opening multiple tabs,
- there are no value types available other than binary and text, and
- the architecture doesn't support hot loading of operations, which limits the scope of what it can offer.

With ciphereditor, I'd like to break out of these boundaries while maintaining user friendliness. This is a real challenge but I'm still convinced that it is worth taking the risk. As ciphereditor is still in the alpha development stage, a lot of changes are yet to come.

cryptii isn't abandoned as I plan to bring the operation library of ciphereditor to cryptii at some point, keeping the simplified sequential UI intact, as an alternative to the full ciphereditor experience.

**Comment by SevHub** [comment]
Thank you for your detailed answer. Now a lot of things are much clearer too me, and now the website is very useful too me. One more question: Does the website also provide AES encryption and decryption? I have not found such a function yet.

**Reply by ffraenz** [reply]
All cryptii operations will be implemented in ciphereditor, eventually. AES is not available right now but will be in the future. Stay tuned!

**Reply by SevHub** [reply]
Hi again, after some time I came back to the website for some tinkering. Again I am stuck with a seemingly simple task (at least in criptii its a matter of seconds to find out how to do it...). I have a string that that is some base64 encoded data. I just want to parse the string and see the data in HEX form. Seems to be impossible in ciphereditor, I can only transform clear text into a base64 representation. The other way I can only get a text representation (what is useless if the data in the base64 string is not text but binary data). Trying to pipe the resulting text into another transformation node suddenly changes the flow direction of the nodes which is extremly frustrating and annoying. 
This in general a criticism I have: For me it is never clear what the direction of data flow is. In criptii its completly clear. In ciphereditor it sometimes changes automatically. It would be better to just show an error (e.g. when I try to take binary (zero and ones) as input, but the input field contains invalid characters) than changing the nodes that are already in the pipeline.
